aggregator/src/aggregator/aggregation_job_writer.rs

```diff
@@ -809,129 +614,161 @@ where
             }
 
             if report_aggregations.iter().all(|ra| ra.is_terminal()) {
-                *aggregation_job = Cow::Owned(
-                    aggregation_job
-                        .as_ref()
-                        .clone()
-                        .with_state(AggregationJobState::Finished),
-                );
+                *aggregation_job.to_mut() = aggregation_job
+                    .as_ref()
+                    .clone()
+                    .with_state(AggregationJobState::Finished);
             }
         }
     }
 
-    /// Update batches to reflect updates to aggregation jobs.
-    fn update_batches<BUC: BatchUpdateCallback>(
-        &mut self,
-        clock: &impl Clock,
-    ) -> Result<(), Error> {
-        // Update in-memory state of batches: outstanding job counts will change based on new or
-        // completed aggregation jobs affecting each batch; the client timestamp interval will
-        // change based on the report aggregations included in the batch.
-        self.batches = self
-            .by_batch_identifier_index
-            .iter()
-            .flat_map(|(batch_identifier, by_aggregation_job_index)| {
-                let (batch_op, mut batch) = match (
-                    BUC::write_type(),
-                    self.batches.remove(batch_identifier),
-                ) {
-                    (_, Some((batch_op, batch))) => (batch_op, batch),
-                    (AggregationJobWriteType::Create, None) => (
+    /// Update batch aggregations to reflect the aggregation jobs that will be written.
+    fn update_batch_aggregations(&mut self) -> Result<(), Error> {
+        let aggregation_parameter = match self.writer.aggregation_parameter() {
+            Some(aggregation_parameter) => aggregation_parameter,
+            // None means there are no aggregation jobs to write, so we can safely short-circuit.
+            None => return Ok(()),
+        };
+
+        for (batch_identifier, by_aggregation_job_index) in &self.writer.by_batch_identifier_index {
+            // Grab the batch aggregation we read for this batch identifier, or create a new empty
+            // one to aggregate into.
```

Comment:
I'm trying to reason through how we preserve the existing logic where we used a batch's `OPEN, CLOSING, CLOSED` state to prevent the leader from creating new aggregation jobs once a collection had been initiated. How is that achieved now?